conan/tools/cmake/toolchain/blocks.py

```diff
@@ -667,6 +667,37 @@ class TryCompileBlock(Block):
         """)
 
 
+class CompilersBlock(Block):
+    template = textwrap.dedent(r"""
+        {% for lang, compiler_path in compilers.items() %}
+        set(CMAKE_{{ lang }}_COMPILER "{{ compiler_path|replace('\\', '/') }}")
+        {% endfor %}
+        {% for lang, compiler_launcher_path in compiler_launchers.items() %}
+        set(CMAKE_{{ lang }}_COMPILER_LAUNCHER "{{ compiler_launcher_path|replace('\\', '/') }}")
+        {% endfor %}
+    """)
+
+    def context(self):
+        # Reading configuration from "tools.build:compiler_executables" -> {"C": "/usr/bin/gcc"}
+        compilers_by_conf = self._conanfile.conf.get("tools.build:compiler_executables", default={},
+                                                     check_type=dict)
+        # Map the possible languages
+        compilers = {}
+        compiler_launchers = {}
+        # Allowed <LANG> variables (and <LANG>_LAUNCHER)
+        compilers_mapping = {"cc": "C", "cuda": "CUDA", "cxx": "CXX",
+                             "objc": "OBJC", "objcxx": "OBJCXX", "rc": "RC"}
+        for comp, lang in compilers_mapping.items():
+            # To set CMAKE_<LANG>_COMPILER
+            if comp in compilers_by_conf:
+                compilers[lang] = compilers_by_conf[comp]
+            # To set CMAKE_<LANG>_COMPILER_LAUNCHER
```

Comment:
I am afraid that ``COMPILER_LAUNCHER`` vars are CMake 3.17: https://cmake.org/cmake/help/latest/envvar/CMAKE_LANG_COMPILER_LAUNCHER.html, so this wouldn't be valid, we have the commitment to support 3.15: https://github.com/conan-io/tribe/pull/4

Reply:
Oh, you're completely right. Changing it